### PR TITLE
Add key management scenarios

### PR DESCRIPTION
**Pull request by mnm678**
Per the discussion in today's Notary v2 meeting I added an initial draft of some key management scenarios.

### CODE REVIEW

File: key-management-scenarios.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Notary Key Management Scenarios
+In order to allow for end to end signing of artifacts, Notary v2 will require the use of cryptographic keys. These scenarios define how these keys will be generated, distributed, and trusted in Notary v2.
+
+## Scenarios
+
+### Scenario #1: A developer signs a package for the first time
+The first time a developer signs a package, they will need to set up a signing key and their local environment.
+  1. Generate a developer signing key.
+  1. Create a local repository that delegates to the developer key notary-tool --create-local-repository, notary-tool --delegate developer-key.pub
```
Comment (SteveLasker) [inline key-management-scenarios.md:9]: Create a local repository is a little specific, and I'm not sure what it means in the sense of a registry. 
Is this:
- Developers provides a signing key, or reference to a signing key - where they get the key is outside the scope
- Developer signs local content

The details for the cli are design choices, not part of the scenario.
We do need to start a UX discussion, which will get shaped shortly as part of a prototyping phase.

File: key-management-scenarios.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+### Scenario #2: A developer is given permission to upload to a registry (for the first time)
+The first time a developer uploads to a registry, they need to establish that they are a trusted developer.
+  1. Add the developer key to delegating metadata on the registry. The delegating metadata may be top-level targets metadata, an organization-hosted targets metadata file (see TAP 13 for how organization-specific targets metadata can be used), or a different targets metadata file.
```
Comment (SteveLasker) [inline key-management-scenarios.md:13]: Line 12 - great.
Again, way too detailed for implementation. 
Can we please avoid specific assumptions about design in the scenarios, such as references to TAP x

File: key-management-scenarios.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+### Scenario #3: A developer updates their signing key
+In the event of a compromised or lost key, a developer may need to switch to a new signing key.
+  1. The developer will generate a new signing key.
+  1. The new developer key will need to be added to the delegating targets metadata.
+  1. The previous developer key should no longer be trusted.
```
Comment (SteveLasker) [inline key-management-scenarios.md:19]: first and 3rd are great. 2nd assumes delegation.

**Comment by mnm678** [comment]
Thanks for the feedback @SteveLasker

I updated the scenarios to make them more design-independent.

File: key-management-scenarios.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+### Scenario #1: A developer signs a package for the first time
+The first time a developer signs a package, they will need to set up a signing key and their local environment.
+  1. Generate a developer signing key.
+  1. Create a local environment that delegates to the developer key for testing.
+  1. The developer signs local content.
+
+### Scenario #2: A developer is given permission to upload to a registry (for the first time)
+The first time a developer uploads to a registry, they need to establish that they are a trusted developer.
+  1. Add the developer key to delegating party on the registry. There may be a single delegating party on a registry, or multiple that are trusted for different images.
```
Comment (SteveLasker) [inline key-management-scenarios.md:14]: nit: ...different images --> different artifacts.

**Pull request closed without merging**